pulsar-common/src/main/java/org/apache/pulsar/common/protocol/schema/SchemaHash.java

```diff
@@ -55,15 +82,123 @@ public static SchemaHash of(SchemaData schemaData) {
     }
 
     public static SchemaHash of(SchemaInfo schemaInfo) {
-        return of(schemaInfo == null ? new byte[0] : schemaInfo.getSchema(),
+        return of(schemaInfo == null ? null : schemaInfo.getSchema(),
                 schemaInfo == null ? null : schemaInfo.getType());
     }
 
-    public static SchemaHash of(byte[] schemaBytes, SchemaType schemaType) {
-        return new SchemaHash(hashFunction.hashBytes(schemaBytes), schemaType);
+    public static SchemaHash empty() {
+        return of(null, null);
+    }
+
+    private static SchemaHash of(byte[] schemaBytes, SchemaType schemaType) {
+        SchemaHash result;
+        if (schemaBytes == null || schemaBytes.length == 0) {
+            result = EmptySchemaHashFactory.get(schemaType);
+            if (result == null) {
```

Comment:
Why should that method possibly fail? Couldn’t we assume that it always returns a result. This would simplify the code here.

Reply:
One possible is that assume we add a new schemaType, but forget to add it in the `EmptySchemaHashFactory `